modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsRepositoryImpl.java

```diff
@@ -45,204 +50,214 @@
      * Constructor.
      *
      * @param storeData If {@code true} - node stores data locally, {@code false} - otherwise.
-     * @param persistence If {@code true} - node have persistence store, {@code false} - otherwise.
-     * @param statisticsManager Ignite statistics manager.
-     * @param log Ignite logger to use.
+     * @param db Database to use in storage if persistence enabled.
+     * @param subscriptionProcessor Subscription processor.
+     * @param statisticsMgr Ignite statistics manager.
+     * @param logSupplier Ignite logger supplier to get logger from.
      */
     public IgniteStatisticsRepositoryImpl(
             boolean storeData,
-            boolean persistence,
-            IgniteStatisticsManagerImpl statisticsManager,
-            IgniteLogger log) {
+            IgniteCacheDatabaseSharedManager db,
+            GridInternalSubscriptionProcessor subscriptionProcessor,
+            IgniteStatisticsManagerImpl statisticsMgr,
+            Function<Class<?>, IgniteLogger> logSupplier
+    ) {
         if (storeData) {
             // Persistence store
-            partsStats = (persistence) ? null : new ConcurrentHashMap<>();
-            localStats = new ConcurrentHashMap<>();
+            store = (db == null) ? new IgniteStatisticsInMemoryStoreImpl(logSupplier) :
```

Comment:
Let's pass correct implementation from outside.

Reply:
But earlier we decided that implementation can be selected by existence of the database. And in which situation we could pass different store here?